thunder/executors/nvfuserex_impl.py

```diff
@@ -383,10 +382,13 @@ def check_dtensor_tracing_and_runtime_metadata(inp):
             lambda: "nvfuser: Expected runtime and tracing metadata to be the same for DTensor.",
         )
 
-        fd = MultiDeviceFusionDefinition(definition, sorted_unique_inputs, max_length=MAX_LENGTH)
+        fd = DirectFusionDefinition()
         # Device may be set in one of the "factory" methods like full, iota, or uniform
         # NOTE: This should be called before defining because a factory method may look-up at `_selected_device` while being defined.
         fd._selected_device = None
+        with fd:
```

Comment:
Unrelated: How is `_selected_device` used in `FusionDefinition`?

Reply:
`_selected_device` is used to set the device for factory ops like `full`, `iota`, etc.

It is optional set in `_select_device`
https://github.com/Lightning-AI/lightning-thunder/blob/29d39bcffa449c72ce9a5dc46be8ce90fbe01cb7/thunder/executors/nvfuserex_impl.py#L1097-L1108

And `_select_device` is called in the lowering for `full`, `iota`, etc.
https://github.com/Lightning-AI/lightning-thunder/blob/29d39bcffa449c72ce9a5dc46be8ce90fbe01cb7/thunder/executors/nvfuserex_impl.py#L1137-L1151